### PR TITLE
Fix problem with returning empty sets in hal+json.

### PR DESCRIPTION
**Pull request by gijskant**
When fetching the list of studies with HTTP header
'Accept: application/hal+json', the application returned code 
406 'Not acceptable' if the set of studies was empty.

### CODE REVIEW

**Comment by cataphract** [comment]
Please fix whatever's wrong in the build (the failure's on `StudyResourceTests`, so it seems pertinent) and add one test of our own. You cannot write a functional test that testes the situation where there are no studies because the test data has studies, but you can test that the link is there.

File: grails-app/controllers/org/transmartproject/rest/StudyController.groovy [hunk 1]
```diff
@@ -51,4 +55,18 @@ class StudyController {
...
+    private def wrapStudies(Object source) {
+        
+        new ContainerResponseWrapper(
+                container: source,
+                componentType: Study,
+                links: [
+                        new Link(grails.rest.render.util.AbstractLinkingRenderer.RELATIONSHIP_SELF,
+                                "/studies"
+                        )
+                ]
```
Comment (cataphract) [inline grails-app/controllers/org/transmartproject/rest/StudyController.groovy]: Style:

* `def` is redundant in `private def` (though it occurs in other methods in rest-api)
* no line after signature
* no need for the constant to be fully qualified and have a long line there. Either static import it or import the class.
* Be consistent in the line breaks. If you put the first argument immediately after the opening parenthesis, then you should put the closing parenthesis after the last argument, with no line break. Symmetry!

**Comment by gijskant** [comment]
@cataphract @forus 
The problem boils down to: 
- not using wrapStudies results in an empty "_links" field in the output when requesting HAL+JSON and in 406 "Not acceptable" when the set of studies is empty.
- using wrapStudies results in the output being a map {"studies": [ ... ]} containing the list of studies, rather than the plain list of studies [ ... ] when requesting for JSON. 

Using wrapStudies results in acceptable output for HAL+JSON, but will change the output for JSON... basically changing the API. I don't know what the effect of that is for external applications. 
The R client expects a map {"studies": [ ... ]}, but I don't know about other applications.

**Comment by forus** [comment]
@gijskant Did fetching studies with `RInterface` work for you? `getStudies()` failed for me on modified version of RESTful API. 

It happens inside following function https://github.com/thehyve/RInterface/blob/master/R/getStudies.R#L27

```
Error in fromJSON(content, handler, default.size, depth, allowComments,  : 
  invalid JSON input
```

**Pull request merged**